File: src/main/java/com/procurement/notice/model/ocds/ElectronicWorkflows.java
package com.procurement.notice.model.ocds;

import com.fasterxml.jackson.annotation.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
        "useOrdering",
        "usePayment",
        "acceptInvoicing"
})
public class ElectronicWorkflows {
    @JsonProperty("useOrdering")
    @JsonPropertyDescription("A True/False field to indicate if electronic ordering will be used. Required by the EU")
    private final Boolean useOrdering;

    @JsonProperty("usePayment")
    @JsonPropertyDescription("A True/False field to indicate if electronic payment will be used. Required by the EU")
    private final Boolean usePayment;

    @JsonProperty("acceptInvoicing")
    @JsonPropertyDescription("A True/False field to indicate if electronic invoicing will be accepted. Required by " +
            "the EU")
    private final Boolean acceptInvoicing;

    @JsonCreator
    public ElectronicWorkflows(@JsonProperty("useOrdering") final Boolean useOrdering,
                               @JsonProperty("usePayment") final Boolean usePayment,
                               @JsonProperty("acceptInvoicing") final Boolean acceptInvoicing) {
        this.useOrdering = useOrdering;
        this.usePayment = usePayment;
        this.acceptInvoicing = acceptInvoicing;
    }
}

File: src/main/kotlin/com/procurement/notice/model/ocds/ElectronicWorkflows.kt
package com.procurement.notice.model.ocds

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

data class ElectronicWorkflows @JsonCreator constructor(

    @get:JsonProperty("useOrdering")
    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val useOrdering: Boolean?,

    @get:JsonProperty("usePayment")
    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val usePayment: Boolean?,

    @get:JsonProperty("acceptInvoicing")
    @field:JsonInclude(JsonInclude.Include.NON_NULL)
    val acceptInvoicing: Boolean?
)
